src/lxc/bdev/lxcbtrfs.c

```diff
@@ -616,8 +616,13 @@ static int btrfs_recursive_destroy(const char *path)
 		ret = ioctl(fd, BTRFS_IOC_TREE_SEARCH, &args);
 		if (ret < 0) {
 			close(fd);
-			ERROR("Error: can't perform the search under %s\n", path);
 			free_btrfs_tree(tree);
+			if (errno == EPERM || errno == EACCES) {
```

Comment:
errno could be modified by the close() or free_btrfs_tree() calls, you probably want it saved right after the ioctl and then look at the saved copy.